src/ch/blinkenlights/android/vanilla/PlaybackService.java

```diff
@@ -2318,17 +2314,26 @@ public int getTimelineLength()
 	}
 
 	/**
-	 * Returns 'Song' with given id from timeline
+	 * Returns {@link Song} with given position from timeline or null if nothing found
 	*/
-	public Song getSongByQueuePosition(int id) {
-		return mTimeline.getSongByQueuePosition(id);
+	public Song getSongByQueuePosition(int pos) {
+		return mTimeline.getSongByQueuePosition(pos);
+	}
+
+	/**
+	 * Retrieve song position in timeline
+	 * @param id song id as defined in {@link Song#id}
+	 * @return absolute position in song timeline or -1 if song currently not in timeline
+	 */
+	public int getQueuePositionForSong(long id) {
+		return mTimeline.getPositionForSong(id);
```

Comment:
Could we rename the function in SongTimeline to have the same function name? (= getQueuePositionForSong)